cmake/alpakaCommon.cmake

```diff
@@ -505,6 +518,21 @@ if(alpaka_ACC_GPU_CUDA_ENABLE)
             if(alpaka_ACC_CPU_B_OMP2_T_SEQ_ENABLE OR alpaka_ACC_CPU_B_SEQ_T_OMP2_ENABLE OR alpaka_ACC_ANY_BT_OMP5_ENABLE)
                 if(NOT MSVC)
                     alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=-fopenmp>)
+
+                    # See https://github.com/alpaka-group/alpaka/issues/1755
+                    if((${CMAKE_CXX_COMPILER_ID} STREQUAL "AppleClang") AND
+                       (${CMAKE_CXX_COMPILER_ID} VERSION_GREATER_EQUAL 13.1.6) AND
+                       (${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS 14.0.0))
+                       message(STATUS "Xcode 13.4 detected. Force-setting OpenMP to version 4.5.")
+                       alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=-fopenmp-version=45>)
+                    endif()
```

Comment:
There is no interaction between the two changes. CMake parses the file line by line and immediately evaluates all function calls and variables it finds. So if we got down here it is already clear that we don't have the evil clang-13 + OMP5 + CUDA combination.

Reply:
Well, I know how cmake works :D

> So if we got down here it is already clear that we don't have the evil clang-13 + OMP5 + CUDA combination.

I find this not so clear. It is clear given the diff, because the relevant code sections are highlighted. But if we change this code in the future again, I am just afraid that we might forget something.

But anyway, I guess we can just move on, because the proposed changes are fine themselves.